src/http/one/Tokenizer.h

```diff
@@ -9,68 +9,52 @@
 #ifndef SQUID_SRC_HTTP_ONE_TOKENIZER_H
 #define SQUID_SRC_HTTP_ONE_TOKENIZER_H
 
-#include "parser/Tokenizer.h"
+#include "parser/forward.h"
+
+class SBuf;
 
 namespace Http {
 namespace One {
 
 /**
- * Lexical processor extended to tokenize HTTP/1.x syntax.
+ * Extracts either an HTTP/1 token or a complete HTTP/1
+ * quoted-string (and sets the quoted accordingly).
+ * Unescapes escaped characters in HTTP/1.1 quoted strings.
+ *
+ * Governed by:
+ *  - RFC 1945 section 2.1
+ *  "
+ *    A string of text is parsed as a single word if it is quoted using
+ *    double-quote marks.
+ *
+ *        quoted-string  = ( <"> *(qdtext) <"> )
+ *
+ *        qdtext         = <any CHAR except <"> and CTLs,
+ *                         but including LWS>
+ *
+ *    Single-character quoting using the backslash ("\") character is not
+ *    permitted in HTTP/1.0.
+ *  "
+ *
+ *  - RFC 7230 section 3.2.6
+ *  "
+ *    A string of text is parsed as a single value if it is quoted using
+ *    double-quote marks.
  *
- * \see ::Parser::Tokenizer for more detail
+ *    quoted-string  = DQUOTE *( qdtext / quoted-pair ) DQUOTE
+ *    qdtext         = HTAB / SP /%x21 / %x23-5B / %x5D-7E / obs-text
+ *    obs-text       = %x80-FF
+ *  "
+ *
+ * \param http1p0 HTTP/1.0 does not permit \-escaped characters
+ * \param tokenPrefixResult function return value when input is a token prefix
+ * \returns tokenPrefixResult if input contains nothing but a token (prefix)
+ * \returns true (and sets the value) if input starts with a token or quoted-string
```

Comment:
We can remove the parens because we document that aspect below now:

```suggestion
 * \returns true if input starts with a token or quoted-string
```

Reply:
Done.